File: programmers/스택&큐/프로세스.py
def solution(priorities, location):
    answer = 0
    maxPlace = priorities.index(max(priorities))

    while True :
        value = max(priorities)

        if (priorities[maxPlace] == value) :
            priorities[maxPlace] = 0
            answer += 1

            if maxPlace == location :
                break
        maxPlace += 1
        if (maxPlace >= len(priorities)) :
            maxPlace = 0
    return answer

print(solution([2, 1, 3, 2], 2)) # 1
print(solution([1, 1, 9, 1, 1, 1], 0)) # 5
# location은 몇번째로 실행되는지 알고싶은 프로세스의 위치(인덱스)

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
import java.io.*;

class Solution {
        public int solution(int[] priorities, int location) {
            int answer = 0;
            PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder()); // 우선순위 큐

            // 값이 큰 기준으로 우선순위 큐 생성
            for (int p : priorities){
                pq.offer(p);
            }

            while (!pq.isEmpty()) {
                // 우선순위 큐의 값과 일치하는 값을 찾기 위해 priorities 배열 돌기
                for (int i=0; i< priorities.length; i++) {
                    // 현재 우선순위 큐의 가장 우선적인 값이 priorities 배열의 i번째 값과 일치할때
                    if (pq.peek() == priorities[i]){
                        pq.poll();
                        answer ++; // 우선순위 큐에서 제거하면서 출력 순서 증가

                        if (location == i) return answer; // 현재 인덱스가 찾아야하는 인덱스와 일치하면 출력 순서 출력
                    }
                }
            }
            return  answer;
        }
    }
